### PR TITLE
Adding an undefined state to progress bar

### PR DESCRIPTION
**Pull request by Vagdish**
by passing "None" as a value.

### CODE REVIEW

File: kivy/uix/progressbar.py [hunk 1]
```diff
@@ -43,10 +44,40 @@ def _get_value(self):
...
+    
+    undefined_period_time = NumericProperty(2.0)
+    '''Time taken (in seconds) by the indicator to do a round-trip.
+    
+    :attr:`undefined_frequency` is an :class:`~kivy.properties.NumericProperty`.
+    Default value: 2 
+    
+    '''
+    
+    undefined_hint_size = BoundedNumericProperty(0.2, min=0, max=1)
```
Comment (tshirtman) [inline kivy/uix/progressbar.py]: BoundedNumericProperty is not imported at this point :)

File: kivy/data/style.kv [hunk 2]
```diff
@@ -70,9 +70,9 @@
             size: self.width, 24
             source: 'atlas://data/images/defaulttheme/progressbar_background'
         BorderImage:
-            border: [int(min(self.width * (self.value / float(self.max)) if self.max else 0, 12))] * 4
-            pos: self.x, self.center_y - 12
-            size: self.width * (self.value / float(self.max)) if self.max else 0, 24
+            border: [int(min(self.width * (self.value / float(self.max)) if (self.max and self.value is not None) else 0, 12))] * 4
+            pos: self.x if self.value is not None else self.x + (sin(self._time*2*pi/self.undefined_period_time) + 1) / 2 * self.width * (1 - self.undefined_hint_size), self.center_y - 12
```
Comment (tshirtman) [inline kivy/data/style.kv]: sin needs to be imported in kv
#:import sin math.sin

**Comment by tshirtman** [comment]
The result is quite nice :)

**Comment by Vagdish** [comment]
Yup, sorry... I did the import in my master branch. I did a mistake while playing with branches.

I added a commit to fix the imports.

**Comment by inclement** [comment]
Thanks for this! I have minor reservations about it but I'll think about it a little, and have no major objections to merging - if someone else wants to merge it, I confirm it works and the code seems good.

**Comment by Vagdish** [comment]
Thanks inclement.
I'm willing to update the pull request according to your reservations. Do not hesitate to share them ;)

**Comment by dessant** [comment]
This would be great to have, although there are some small docstring issues and we could also tweak the property names a little.

For a start, versionadded tags should be added for each new prop.

**Comment by dessant** [comment]
@inclement do you have a suggestion for the property names?

**Comment by Vagdish** [comment]
Property names cause some trouble here.
I proposed a new pull request #3378 to get rid of property names. Visuals can be changed through style.kv though.
Sorry for the "double request".

**Comment by KeyWeeUsr** [comment]
It was moved to #3378 and the "repo"(branch rather?) doesn't exist anyway.

**Pull request closed without merging**